src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -296,6 +296,191 @@ EnsureModificationsCanRun(void)
 }
 
 
+/*
+ * IsNonDistributedTable returns whether the given relation ID identifies a
+ * non-distributed table such as reference table.
+ */
+bool
+IsNonDistributedTable(Oid relationId)
+{
+	if (!IsCitusTable(relationId))
+	{
+		return false;
+	}
+	CitusTableCacheEntry *tableEntry = GetCitusTableCacheEntry(relationId);
+	return IsNonDistributedTableCacheEntry(tableEntry);
+}
+
+
+/*
+ * IsReferenceTable returns whether the given relation ID identifies a reference
+ * table.
+ */
+bool
+IsReferenceTable(Oid relationId)
+{
+	if (!IsCitusTable(relationId))
+	{
+		return false;
+	}
+	CitusTableCacheEntry *tableEntry = GetCitusTableCacheEntry(relationId);
+	return IsReferenceTableCacheEntry(tableEntry);
+}
+
+
+/*
+ * IsDistributedTable returns whether the given relation ID identifies a distributed
+ * table. A distributed table can be of hash, range or append type.
+ */
+bool
+IsDistributedTable(Oid relationId)
+{
+	if (!IsCitusTable(relationId))
```

Comment:
We can return the cache entry from a variation of IsCitusTable to reduce the call.

I thought that the actual cost of calling LookupCitusTableCacheEntry is going to be low because we will do a simple lookup in most of the cases from a hash table. However, I havent don't any benchmark, do you have any data on this?

Reply:
I don't have any data, that's why I'm trying to be cautious.